private/pkg/oauth2/client_test.go

```diff
@@ -87,7 +87,7 @@ func TestRegisterDevice(t *testing.T) {
 				Body:       io.NopCloser(strings.NewReader(`server error`)),
 			}, nil
 		},
-		err: fmt.Errorf("oauth2: invalid response: 500 server error"),
+		err: fmt.Errorf("oauth2: %w: 500 server error", ErrUnsupported),
```

Comment:
Nit, should these set of tests be referencing HTTP 501 - Not Implemented?

Reply:
Yep, will be a 501 when intentionally disabling it, but I think old servers would 404 too. This test is just showing the error message is captured in the error.